src/connector/src/source/filesystem/opendal_source/mod.rs

```diff
@@ -53,6 +54,9 @@ pub struct GcsProperties {
 
     #[serde(flatten)]
     pub unknown_fields: HashMap<String, String>,
+
+    #[serde(rename = "compression_format")]
+    pub compression_format: Option<DecompressionFormat>,
```

Comment:
Given that we have `None` compression format defined in the enum, I think we don't need the `Option` here. We can just  let the `compression_format` default to `CompressionFormat::None` if not specified.